tests/e2e/specs/modules/pagespeed-insights/activation.test.js

```diff
@@ -6,19 +6,25 @@ import { visitAdminPage, activatePlugin } from '@wordpress/e2e-test-utils';
 /**
  * Internal dependencies
  */
-import { resetSiteKit, deactivateAllOtherPlugins } from '../../../utils';
+import {
+	deactivateAllOtherPlugins,
+	pasteText,
+	resetSiteKit,
+	setSearchConsoleProperty,
+	setSiteVerification,
+} from '../../../utils';
 
 describe( 'PageSpeed Insights Activation', () => {
 	beforeEach( async() => {
 		await activatePlugin( 'e2e-tests-auth-plugin' );
-		await activatePlugin( 'e2e-tests-site-verification-plugin' );
+		await setSiteVerification();
```

Comment:
Hmm, you replaced the utility plugin with setting that value via JS - why that? This ties in with my other comments, I think such base requirements should happen via utility plugin, unless for tests in which these actions specifically are tested.

Reply:
One reason is for speed. Activating utility plugins happens one at a time and is rather slow compared to the API-based utility, especially when running the test over and over in development. Perhaps a utility plugin may be the proper implementation here, but ideally we would have a single plugin for skipping/faking the setup, which we don't (although I suppose we could). Each piece of the setup has been split into separate utility plugins due to the need to activate some pieces while not activating others. Even the auth plugin is doing too much IMO and should probably be split into 2 or 3 pieces.

The setup utility plugins (auth + site verification) are currently filter-based which makes them only applicable to the current user. This makes them unusable for multi-user tests and requires that we have utilities to actually set the data in the DB which is what the REST utilities do. They don't have to be filter-based of course, but then why use a utility plugin for setting data on activation when we have a persistent REST route for doing the same thing while providing much more control.

My understanding and approach recently has been that we were going to leverage the REST based utilities more and use utility plugins for cases which were more test-specific where needed.

There's clearly more than one way to go about this so perhaps we should discuss this a bit more to clarify the direction going forward 😄

Reply:
One reason is for speed. Activating utility plugins happens one at a time and is rather slow compared to the API-based utility, especially when running the test over and over in development. Perhaps a utility plugin may be the proper implementation here, but ideally we would have a single plugin for skipping/faking the setup, which we don't (although I suppose we could). Each piece of the setup has been split into separate utility plugins due to the need to activate some pieces while not activating others. Even the auth plugin is doing too much IMO and should probably be split into 2 or 3 pieces.

The setup utility plugins (auth + site verification) are currently filter-based which makes them only applicable to the current user. This makes them unusable for multi-user tests and requires that we have utilities to actually set the data in the DB which is what the REST utilities do. They don't have to be filter-based of course, but then why use a utility plugin for setting data on activation when we have a persistent REST route for doing the same thing while providing much more control.

My understanding and approach recently has been that we were going to leverage the REST based utilities more and use utility plugins for cases which were more test-specific where needed.

There's clearly more than one way to go about this so perhaps we should discuss this a bit more to clarify the direction going forward 😄